zyte_common_items/pages.py

```diff
@@ -2,23 +2,39 @@
 
 import attrs
 from web_poet import ItemPage, RequestUrl, Returns, WebPage, field
-from web_poet.fields import FieldsMixin
+from web_poet.pages import ItemT
 
 from .components import Metadata
 from .items import BusinessPlace, Product, ProductList, RealEstate
+from .util import format_datetime
 
 
-class _BaseMixin(FieldsMixin):
+def _date_downloaded_now():
+    return format_datetime(datetime.utcnow())
+
+
+class _BasePage(ItemPage[ItemT]):
     @field
     def metadata(self) -> Metadata:
         return Metadata(
-            dateDownloaded=f"{datetime.utcnow().isoformat(timespec='seconds')}Z",
+            dateDownloaded=_date_downloaded_now(),
             probability=1.0,
         )
 
+    def no_item_found(self) -> ItemT:
+        """Return an item with the current url and probability=0,
+        indicating that the passed URL doesn't contain the expected item.
+
+        Use it in your .validate_input implementation.
+        """
+        return self.item_cls(  # type: ignore
+            url=self.url,  # type: ignore[attr-defined]
+            metadata=Metadata(probability=0, dateDownloaded=_date_downloaded_now()),
```

Comment:
Maybe we could include an example with BusinessPlace (which uses BusinessPlaceMetadata). I think it will work, since BusinessPlaceMetadata is a Metadata subclass, but I think covering it in the examples makes sense nonetheless.

Reply:
Are you ok with merging it as-is @Gallaecio?